pkg/scheduler/factory/factory.go

```diff
@@ -1064,6 +1067,72 @@ func (c *configFactory) invalidateCachedPredicatesOnNodeUpdate(newNode *v1.Node,
 	}
 }
 
+func nodeSchedulingPropertiesChanged(newNode *v1.Node, oldNode *v1.Node) bool {
+	if nodeAllocatableChanged(newNode, oldNode) {
+		klog.V(4).Infof("Allocatable resource of node %s changed", newNode.Name)
+		return true
+	}
+	if nodeLabelsChanged(newNode, oldNode) {
+		klog.V(4).Infof("Labels of node %s changed", newNode.Name)
+		return true
+	}
+	if nodeTaintsChanged(newNode, oldNode) {
+		klog.V(4).Infof("Taints of node %s changed", newNode.Name)
+		return true
+	}
+	if nodeConditionsChanged(newNode, oldNode) {
+		klog.V(4).Infof("Conditions of node %s changed", newNode.Name)
+		return true
+	}
+	if newNode.Spec.Unschedulable != oldNode.Spec.Unschedulable && newNode.Spec.Unschedulable == false {
+		klog.V(4).Infof("Node %s changed to schedulable", newNode.Name)
+		return true
+	}
+	return false
+}
+
+func nodeAllocatableChanged(newNode *v1.Node, oldNode *v1.Node) bool {
+	return !reflect.DeepEqual(oldNode.Status.Allocatable, newNode.Status.Allocatable)
+}
+
+func nodeLabelsChanged(newNode *v1.Node, oldNode *v1.Node) bool {
+	return !reflect.DeepEqual(oldNode.GetLabels(), newNode.GetLabels())
+}
+
+func nodeTaintsChanged(newNode *v1.Node, oldNode *v1.Node) bool {
+	if !reflect.DeepEqual(newNode.Spec.Taints, oldNode.Spec.Taints) {
+		return true
+	}
+	oldTaints, oldErr := helper.GetTaintsFromNodeAnnotations(oldNode.GetAnnotations())
+	if oldErr != nil {
+		// If parse old node's taint annotation failed, we assume node's taint changed.
+		klog.Errorf("Failed to get taints from annotation of old node %s: %v", oldNode.Name, oldErr)
+		return true
+	}
+	newTaints, newErr := helper.GetTaintsFromNodeAnnotations(newNode.GetAnnotations())
```

Comment:
`GetTaintsFromNodeAnnotations` is only used for validation right now; if we did not cherry pick this PR, it's not necessary to me. 

BTW, Toleration/Taint should be GAed for a while, we should remove the annotation :)

Reply:
Yeah, It's not appropriate to use `GetTaintsFromNodeAnnotations` here. I'm not sure if we need to be compatible with old nodes has taint annotations, if not, remove the annotation comparison LGTM.

Reply:
Done